File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        let set= new Set(nums)
        let max=0, count=0 , n=nums.length
        for(let i=0;i<n;i++)
        {   
            let num=nums[i]
            if(!set.has(num-1))
            {
                count++
                while(set.has(num+1))
                {   
                    num++
                    count++
                }
                max=Math.max(max,count)
            }
            count=0

        }
        return max
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.go
func subarraySum(nums []int, k int) int {
	sum,count,i,j:=0,0,0,0

	for {
		
		if j==len(nums) {
			i++
			j=i
			sum=0
		}
		if i==len(nums) {
			break
		}
		sum+=nums[j]
		if sum==k {
			count++
		}
		j++
	}
	return count
}
